src/recipe.js

```diff
@@ -1,16 +1,20 @@
 class Recipe {
-  constructor(recipe) {
+  constructor(recipe, ingredientData) {
     this.id = recipe.id;
     this.name = recipe.name;
     this.image = recipe.image;
     this.tags = recipe.tags;
     this.ingredients = recipe.ingredients;
+    this.ingredientData = ingredientData.filter(data => this.ingredients.some(ingredient => ingredient.id === data.id))
   }
+  
   calculateIngredientsCost() {
-    // return this.ingredients.map(i => {
-    //   ingredientData.find(ingredient => ingredient === i);
-    // });
+    const sortedIngredients = this.ingredients.sort((a, b) => a.id - b.id)
```

Comment:
This sort method of yours saved the day, honestly, it was a pleasure to work on this one with you. Brilliant.